src/components/MessageList/VirtualizedMessageList.js

```diff
@@ -108,15 +104,46 @@ const VirtualizedMessageList = ({
           message={message}
           groupedByUser={
             shouldGroupByUser &&
-            i > 0 &&
-            message.user.id === messageList[i - 1].user.id
+            streamMessagesIndex > 0 &&
+            message.user.id === messageList[streamMessagesIndex - 1].user.id
           }
         />
       );
     },
-    [MessageDeleted, customMessageRenderer, shouldGroupByUser],
+    [
+      MessageDeleted,
```

Comment:
can probably remove MessageDeleted from dep array, it shouldn't change

Reply:
Not sure if I'd remove this. Even if it shouldn't change this, it technically could, and it doesn't really hurt to have it in the dependencies array.

Reply:
I think this raises a larger question of how we handle dependency arrays. On the react native side we only include the data that should actually trigger a new render during an app session, so we don't include things like prop UI components that don't change mid-session. Obviously native is a little different, but we definitely saw performance improvements and decreased re-renders upon getting stricter with the dependency arrays. @mahboubii do you have thoughts on whether or not to keep UI components in dependency arrays?

Reply:
There could be customers who change the UI props conditionally in their app but the performance gain from removing this dependency check is worth it IMO and we should move toward that path. 
In the meantime since changing these effect dependencies are potentially a breaking change, let's leave it as is until the next major release.
we should make it clear for customers that UI prop overrides should not change after the first render of the app

Reply:
Yeah, I added this to the component:
```
// TODO: split scrollSeekPlaceholder into two props (e.g. ScrollSeekPlaceholder and scrollSeekConfiguration) when making breaking changes
```

and this to the typedefs:
```
*  Note: virtuoso has broken out the placeholder value and instead includes it in its components prop.
*  TODO: break out placeholder when making other breaking changes.
```